src/astro/utils/database.py

```diff
@@ -0,0 +1,86 @@
+from typing import Union
+
+from airflow.hooks.base import BaseHook
+from airflow.providers.sqlite.hooks.sqlite import SqliteHook
+from sqlalchemy import create_engine, text
+from sqlalchemy.engine import Engine
+from sqlalchemy.engine.result import ResultProxy
+
+from astro.constants import Database
+from astro.utils.dependencies import BigQueryHook, PostgresHook, SnowflakeHook
+
+
+def get_database_name(
+    interface: Union[Engine, SqliteHook, BigQueryHook, PostgresHook, SnowflakeHook]
```

Comment:
I attempted to have a granular type hint here, using:
`interface: Union[Engine, SqliteHook, BigQueryHook, PostgresHook, SnowflakeHook]`
However, this conflicts with our current strategy on handling dependency:
```
TypeError: Union[arg, ...]: each arg must be a type. Got <astro.utils.dependencies.MissingPackage object at 0x7f350c80e790>.
```
I'll keep `SqliteHook` and  `BaseHook`, removing the others.

Reply:
Oh yikes good to know